### ISSUE TITLE
alot crashes on thread with missing toplevel message

### ISSUE THREAD

**Issue by mjg** [pazz/alot#1243]
alot 0.7-21-gbdd2a51 (master)
Python 2.7.15
notmuch 0.25

Observed behavior
-----------------

`alot search` crashes when a specific thread is included in the search, does not crash otherwise. That thread (a zendesk repsonse) has `In-Reply-To:` pointing to a message that is not in my db:

```
  File "/home/mjg/.local/lib/python2.7/site-packages/alot-0.7-py2.7.egg/alot/db/thread.py", line 246, in get_messages
    for m in thread.get_toplevel_messages():
  File "/usr/lib/python2.7/site-packages/notmuch/thread.py", line 141, in get_toplevel_messages
    raise NullPointerError()
notmuch.errors.NullPointerError
```
After this, alot hangs and can't be `kill`ed, only `kill -9`ed.

I can reproduce this from the python shell in the sense that `t.get_toplevel_messages()` throws `NullPointerError` on that thread, which one may argue is correct (toplevel not present) or not (should return the top of the messages in the db). Also:

```
notmuch show  thread:000000000003b3a4
Internal error: Thread 000000000003b3a4 has no toplevel messages.
 (notmuch-show.c:1016)
```

Apparantly, this will be fixed in notmuch 0.26.2 (see the[notmuch-ml](https://www.mail-archive.com/notmuch@notmuchmail.org/msg46352.html)) which hasn't hit most distros yet.

In any case, I think alot should catch the exception.

Expected behavior
-----------------

Don't crash, display the thread :)

**Comment by mjg** [comment]
In addition, I can confirm that everything is fine when alot is used with notmuch 0.26.2. It returns the topmost of the present messages as toplevel_messages (even though they point non-present messages higher up in the chain). Still, alot shouldn't crash hard.

**Comment by pazz** [comment]
yes I agree that I'd be better to catch this exception than to rely on an updated notmuch version.
I wonder why this fails at said point and not already in the [constructor here](https://github.com/pazz/alot/blob/master/alot/widgets/thread.py#L322).

Anyways, I'm open for PRs. It should catch the exception (and add an inline comment why) in those two positions.
